frontend/document/credocument.lua

```diff
@@ -283,10 +288,173 @@ function CreDocument:updateColorRendering()
     end
 end
 
+function CreDocument:setHideNonlinearFlows(hide_nonlinear_flows)
+    if hide_nonlinear_flows ~= self.hide_nonlinear_flows then
+        self.hide_nonlinear_flows = hide_nonlinear_flows
+        self._document:setIntProperty("crengine.doc.nonlinear.pagebreak.force", self.hide_nonlinear_flows and 1 or 0)
+        self:cacheFlows()
+    end
+end
+
 function CreDocument:getPageCount()
     return self._document:getPages()
 end
 
+-- Whether the document has any non-linear flow to care about
+function CreDocument:hasNonLinearFlows()
+    return self._document:hasNonLinearFlows()
+end
+
+-- Whether non-linear flows (if any) will be hidden
+function CreDocument:hasHiddenFlows()
+    return self.flows[1] ~= nil
+end
+
+-- Get the next/prev page number, skipping non-linear flows,
+-- i.e. the next/prev page that is either in the current
+-- flow or in the linear flow (flow 0)
+-- If "page" is 0, these give the initial and final linear pages
+function CreDocument:getNextPage(page)
+    if self:hasHiddenFlows() then
+        if page < 0 or page >= self:getPageCount() then
+            return 0
+        elseif page == 0 then
+            return self.getLastLinearPage()
+        end
```

Comment:
In onGoToBegenning, you provide page=0 to go to the first page - which sounds like it's not what you do :)

Reply:
Of course, I got them mixed.